R/pkg/R/sparkR.R

```diff
@@ -270,27 +291,97 @@ sparkRSQL.init <- function(jsc = NULL) {
 #'}
 
 sparkRHive.init <- function(jsc = NULL) {
-  if (exists(".sparkRHivesc", envir = .sparkREnv)) {
-    return(get(".sparkRHivesc", envir = .sparkREnv))
+  .Deprecated("sparkR.session.getOrCreate")
+
+  if (exists(".sparkRsession", envir = .sparkREnv)) {
+    return(get(".sparkRsession", envir = .sparkREnv))
   }
 
-  # If jsc is NULL, create a Spark Context
-  sc <- if (is.null(jsc)) {
-    sparkR.init()
-  } else {
-    jsc
+  # Default to without Hive support for backward compatibility.
+  sparkR.session.getOrCreate(enableHiveSupport = TRUE)
+}
+
+#' Get the existing SparkSession or initialize a new SparkSession.
+#'
+#' Additional Spark properties can be set (...), and these named parameters take priority over
+#' over values in master, appName, named lists of sparkConfig.
+#'
+#' @param master The Spark master URL
+#' @param appName Application name to register with cluster manager
+#' @param sparkHome Spark Home directory
+#' @param sparkConfig Named list of Spark configuration to set on worker nodes
+#' @param sparkExecutorConfig Named list of Spark configuration to be used when launching executors
```

Comment:
I think this comes from the time we wanted to support all the features of SparkConf [1]. My recommendation would be to keep the same config name `sparkExecutorEnv` (to minimize confusion / change) and just change the documentation to indicate this is an optional argument to set any environment variables in the executor. 

[1] https://github.com/apache/spark/blob/7c6c6926376c93acc42dd56a399d816f4838f28c/core/src/main/scala/org/apache/spark/SparkConf.scala#L129